### PR TITLE
WSL: Get drives mount point from wsl.conf

### PR DESCRIPTION
**Pull request by kalexey89**
The drives mount point for WSL defined as constant at now. See issue #218. 
This PR will fix it by getting drives mount point dynamically from `wsl.conf` file.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,10 +7,35 @@ const isWsl = require('is-wsl');
...
+ *
+ * @async
+ * @inner
+ * @returns {string} Mount point
+ */
+const getWslMountPoint = async () => {
+	const confFile = '/etc/wsl.conf';
+
+	if (fs.existsSync(confFile)) {
+		const conf = await pReadFile(confFile, {encoding: 'utf-8'});
```
Comment (anaisbetts) [inline index.js:26]: Just use fs.readFileSync here, it's super small. If you want to you could also memoize this whole function

File: index.js [hunk 2]
```diff
@@ -7,10 +7,35 @@ const isWsl = require('is-wsl');
...
+ * Returns the mount point for fixed drives in WSL
+ *
+ * @async
+ * @inner
+ * @returns {string} Mount point
+ */
+const getWslMountPoint = async () => {
+	const confFile = '/etc/wsl.conf';
+
+	if (fs.existsSync(confFile)) {
```
Comment (anaisbetts) [inline index.js:25]: Maybe flip the condition and early-exit so the entire function isn't in an `if` block

**Comment by anaisbetts** [comment]
Seems reasonable to me /cc @sindresorhus

**Comment by kalexey89** [comment]
@anaisbetts I improved source code of function according to your comments. Please check changes.

File: index.js [hunk 3]
```diff
@@ -11,6 +11,38 @@ const pAccess = promisify(fs.access);
 // Path to included `xdg-open`.
 const localXdgOpenPath = path.join(__dirname, 'xdg-open');
 
+/**
+Get the mount point for fixed drives in WSL.
+
+@inner
+@returns {string} The mount point.
+*/
+const getWslDrivesMountPoint = (() => {
```
Comment (sindresorhus) [inline index.js:20]: The `open` function is async, so it should not contain any synchronous IO. That would be unexpected for consumers. You can use `util.promisify` on the `fs` methods.

File: index.js [hunk 4]
```diff
@@ -11,6 +11,38 @@ const pAccess = promisify(fs.access);
...
+	let mountPoint = null;
```
Comment (sindresorhus) [inline index.js:21]:
```suggestion
	let mountPoint;
```

I prefer to use `undefined`.

File: index.js [hunk 5]
```diff
@@ -11,6 +11,38 @@ const pAccess = promisify(fs.access);
...
+
+		const confFile = '/etc/wsl.conf';
+
+		if (!fs.existsSync(confFile)) {
+			// Default value for "root" param
+			// according to https://docs.microsoft.com/en-us/windows/wsl/wsl-config
+			return '/mnt/';
+		}
+
+		const confContent = fs.readFileSync(confFile, {encoding: 'utf-8'});
```
Comment (sindresorhus) [inline index.js:37]:
```suggestion
		const confContent = fs.readFileSync(confFile, {encoding: 'utf8'});
```

File: index.js [hunk 6]
```diff
@@ -11,6 +11,38 @@ const pAccess = promisify(fs.access);
...
+const getWslDrivesMountPoint = (() => {
+	let mountPoint = null;
+
+	return function () {
+		if (mountPoint) {
+			// Return memoized mount point value
+			return mountPoint;
+		}
+
+		const confFile = '/etc/wsl.conf';
```
Comment (sindresorhus) [inline index.js:29]:
```suggestion
		const configFilePath = '/etc/wsl.conf';
```

File: index.js [hunk 7]
```diff
@@ -11,6 +11,38 @@ const pAccess = promisify(fs.access);
...
+		if (!fs.existsSync(confFile)) {
+			// Default value for "root" param
+			// according to https://docs.microsoft.com/en-us/windows/wsl/wsl-config
+			return '/mnt/';
+		}
+
+		const confContent = fs.readFileSync(confFile, {encoding: 'utf-8'});
+
+		mountPoint = (/root\s*=\s*(.*)/g.exec(confContent)[1] || '').trim();
+		mountPoint = mountPoint.endsWith('/') ? mountPoint : mountPoint.concat('/');
```
Comment (sindresorhus) [inline index.js:40]: Use `+`, not `concat`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (kalexey89) [reply index.js:21]: I made this function synchronous according to the previous review (see [comment](https://github.com/sindresorhus/open/pull/219#discussion_r570356185)).

File: index.js [hunk 8]
```diff
@@ -7,10 +7,43 @@ const isWsl = require('is-wsl');
...
+
+	return async function () {
+		if (mountPoint) {
+			// Return memoized mount point value
+			return mountPoint;
+		}
+
+		const configFilePath = '/etc/wsl.conf';
+
+		if (!fs.existsSync(configFilePath)) {
```
Comment (sindresorhus) [inline index.js:32]: This should also be async. And should use `fs.access`.

**Pull request merged**